subtests/docker_cli/run_volumes_selinux/run_volumes_selinux.py

```diff
@@ -72,27 +75,32 @@ def init_container(self, volume=None, volumes_from=None):
         self.sub_stuff['fds'].append(write_fd)
         self.sub_stuff['fds'].append(read_fd)
         dkrcmd = dockercmd.AsyncDockerCmd(self, 'run', subargs)
-        # TODO: Fix use of dkrcmd.stdin when appropriate class mech. available
         dkrcmd.execute(read_fd)
         dkrcmd.stdin = write_fd
         os.close(read_fd)  # no longer needed
-        os.write(write_fd, 'echo "Started"\n')
-        self.failif(not wait_for_output(lambda: dkrcmd.stdout,
-                                        'Started',
-                                        timeout=20),
-                    "Error starting container %s: %s" % (name, dkrcmd))
+
+        # This could take a while if image needs pulling
+        find_name = lambda: str(dc.list_containers_with_name(name))
+        is_running = wait_for_output(find_name, name,
+                                     timeout=self.config['docker_timeout'])
```

Comment:
I'm having a tough time with this. First, understanding it (but I finally did). Then, making sure it does what it's supposed to (which I still haven't confirmed). The unnecessary duplicate use of `name` troubles me. Is there any reason not to just use

    find_name = lambda: dc.list_containers_with_name(name) != []
    utils.wait_for(find_name, timeout=...) ?